### DISCUSSION TITLE
Summary of H5P Content Types in a Title

### DISCUSSION THREAD

**Discussion by cogdog** [pressbooks/ideas#487] [Ideas]
We have a project where there is interest in knowing how many different content types of H5P are used in a title and ideally, how many items of each type, some kind of frequency data table like:

> - 15 Accordions
> - 23 Question Sets
> - 2 Branching Scenerios

The /h5p-listing URL seems to be pulling this data as it lists the H5P type for each single item, so it looks like functionality is tere to get info on each H5P item. Or is there an API endpoint? I know only the one that returns all the H5P urls n a title.

**Comment by richard015ar** [comment]
Hi @cogdog, this is something totally doable. At the moment the H5P plugin does not offer an API endpoint with more details of each H5P activity, other than the URL, as you mentioned.

In order to do that, a new endpoint needs to be created. All the information needed is under two DB Tables:
- `wp_<BLOG_ID>_h5p_contents` (to get the H5P titles from a book)
- `wp_<BLOG_ID>_h5p_libraries` (to get the types -- look for the `name` column).

In the new WP API endpoint all the logic/grouping/parameter/filtering needs to be implemented to display the information requested.

It's an interesting project, I look forward to having some time to work on this.
